### PR TITLE
Add Disable Auth Feature Flag

### PR DESCRIPTION
**Pull request by frank-kerrigan**
Add Auth0 feature flag so it can run without Auth.

The feature flag does not have to added to appsettings as it will default to Auth Enabled.

### CODE REVIEW

File: src/DigitalPreservation/DigitalPreservation.UI/Program.cs [hunk 1]
```diff
@@ -29,24 +27,32 @@
...
-    builder.Services.AddRazorPages().AddMvcOptions(options =>
-    {
-        var policy = new AuthorizationPolicyBuilder()
-            .RequireAuthenticatedUser()
-            .Build();
-        options.Filters.Add(new AuthorizeFilter(policy));
-    }).AddMicrosoftIdentityUI();
+
+        // <ms_docref_add_default_controller_for_sign-in-out>
+        builder.Services.AddRazorPages().AddMvcOptions(options =>
```
Comment (donaldgray) [inline src/DigitalPreservation/DigitalPreservation.UI/Program.cs:48]: ~Do you still need to call `.AddRazorPages()` regardless of whether there is auth or not?~

Edit - `.AddRazorPages()` is called at line 70 - might be worth the call in 1 place only?

File: src/DigitalPreservation/DigitalPreservation.UI/Program.cs [hunk 2]
```diff
@@ -88,9 +94,12 @@
     app.MapControllers();
     app.UseHealthChecks("/health");
     //Authentication
-    app.UseAuthentication();
-    app.UseAuthorization();
-   
+    if (useAuthFeatureFlag)
+    {
+        app.UseAuthentication();
+        app.UseAuthorization();
```
Comment (donaldgray) [inline src/DigitalPreservation/DigitalPreservation.UI/Program.cs]: `.UseAuthorization()` is called regardless as it's on line 91

**Review by donaldgray — APPROVED** [review]

File: src/DigitalPreservation/DigitalPreservation.UI/Program.cs (hunk 1 above)
Comment (frank-kerrigan) [reply src/DigitalPreservation/DigitalPreservation.UI/Program.cs:48]: fixed

File: src/DigitalPreservation/DigitalPreservation.UI/Program.cs (hunk 2 above)
Comment (frank-kerrigan) [reply src/DigitalPreservation/DigitalPreservation.UI/Program.cs]: yes, removed line

**Pull request merged**